providers/flagd/pkg/provider.go

```diff
@@ -13,6 +13,7 @@ import (
 )
 
 type Provider struct {
+	initialized           bool
```

Comment:
maybe we can be more explicit and pass a false value into the `NewProvider` func

Reply:
We don't have to pass the value for `initialized` state through `NewProvider`as it's not up to the end user to set it.
But I explicitly assigned a `false` at struct initialization time.

Reply:
Sorry, I wrote it poorly. What you've done is exactly what I was suggesting :)